bin/completed_segments.py

```diff
@@ -0,0 +1,77 @@
+#!/usr/bin/env python
+""" Find (and delete) fully completed kafka-backup segments """
+
+import os
+import re
+import sys
+import argparse
+
+
+def filename_pattern():
+    """ Kafka-backup segment/index filename pattern """
+    return 'segment_partition_%s_from_offset_%s_%s'
+
+
+def parse_args():
+    """ Parse cmdline args """
+    parser = argparse.ArgumentParser(
+        description='Find (and delete) fully completed kafka-backup segments'
+    )
+    parser.add_argument(
+        '-d', '--delete',
+        help='Delete fully completed segment files',
+        action='store_true',
+        default=False,
+    )
+    parser.add_argument(
+        'target_dir',
+        help='Backup directory (target.dir)',
+        default=os.getcwd(),
+    )
+    return parser.parse_args()
+
+
+def collect_segments(target_dir):
+    """ Collect segment partitions & offsets per directory (topic) """
+    # Make regex from pattern
+    fregex = re.compile(r"^%s$" % (filename_pattern() % (r'(\d{3})', r'(\d{10})', 'records')))
+    res = {}
+    # Traverse dirtree to collect offsets in partitions per topic
+    for tdir, _, files in os.walk(target_dir):
+        if tdir == target_dir:
+            continue
+        res[tdir] = {}
+        for segfile in sorted(files):
+            match = fregex.match(segfile)
+            if match:
+                partition = match.group(1)
```

Comment:
i prefer the `(partition,offset)=match.groups()` syntax. but up to you ;)

Reply:
ah, nice! Missed this syntax is applicable here as well